aptos-move/framework/src/natives/object_code_deployment.rs

```diff
@@ -0,0 +1,24 @@
+// Copyright © Aptos Foundation
+// SPDX-License-Identifier: Apache-2.0
+
+use move_core_types::account_address::AccountAddress;
+use serde::{Deserialize, Serialize};
+
+#[derive(Debug, Serialize, Deserialize, Eq, PartialEq)]
+pub struct ExtendRef {
+    #[serde(rename = "self")]
+    pub address: AccountAddress,
+}
+
+#[derive(Debug, Serialize, Deserialize, Eq, PartialEq)]
+pub struct PublisherRef {
+    pub extend_ref: ExtendRef,
+}
+
+impl PublisherRef {
```

Comment:
Why do we need these?

Reply:
This is to deserialize the response when retrieving the `PublisherRef` in the tests. We validate that the `ExtendRef` in the `PublisherRef` contains the object address as expected.